### PR TITLE
DROOLS-1828: [Guided Decision Table] New Toolbar

### PR DESCRIPTION
**Pull request by karreiro**
See:
- https://issues.jboss.org/browse/RHBRMS-2946
- https://issues.jboss.org/browse/DROOLS-1828

---

demo

---

Part of an ensemble:
- https://github.com/errai/errai/pull/303
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/617
- https://github.com/AppFormer/uberfire/pull/913
- https://github.com/kiegroup/drools-wb/pull/708

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@karreiro Not reviewed as such atm.. just playing. I notice that the "Configure the columns first, then add rows (rules)..." warning does not disappear for new tables (with no columns) that are sub-sequentially added. This is also evident in your video. Should it disappear once a column has been added? (I suspect it should but thought I'd ask first).

**Comment by karreiro** [comment]
@manstis Good catch! Fixing that..

**Comment by manstis** [comment]
@karreiro Please let me know when you've updated and I'll take a look.

**Comment by karreiro** [comment]
@manstis Changes applied. Could you please take a look? :-)

gif-2

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java [hunk 1]
```diff
@@ -286,6 +287,55 @@ protected void activateDocument(final GuidedDecisionTableView.Presenter dtPresen
                          dtPresenter.getDataModelOracle(),
                          dtPresenter.getModel().getImports(),
                          !dtPresenter.getAccess().isEditable());
+
+        addColumnsTab();
+
+        columnsTabToggle(dtPresenter);
+    }
+
+    void columnsTabToggle(final GuidedDecisionTableView.Presenter decisionTablePresenter) {
```
[the remark this answers is not in the document]
Comment (karreiro) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java]: @jomarko Good catch! Fixed:
gif-5

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/resources/i18n/GuidedDecisionTableConstants.java [hunk 2]
```diff
@@ -362,9 +362,7 @@ String AuditLogEntryOn1(String date,
 
     String NoColumnsAvailable();
 
-    String AddColumn();
-
-    String EditColumns();
-
     String TablePinnedModeHelp();
+
+    String PinnedMode();
```
[the remark this answers is not in the document]
Comment (karreiro) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/resources/i18n/GuidedDecisionTableConstants.java]: Fixed.. (I slightly changed the sentence):
gif-3

[the remark this answers is not in the document]
Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/resources/i18n/GuidedDecisionTableConstants.java]: Sorry, I know it is detail, but the sentence is still not true.

- Enter pin mode (pin mode indicator is shown)
- Point mouse to table name (Pricing loans in this case)
- Press and hold left mouse button
- Drag mouse
- The table will be moving. 

Either it is issue or the sentence should be updated, let me know what do you think.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java]: Looks cool, thanks

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/resources/i18n/GuidedDecisionTableConstants.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (karreiro) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/resources/i18n/GuidedDecisionTableConstants.java]: You're right, Jozef. I just applied the original sentence.

**Comment by manstis** [comment]
I have a couple of concerns; I am checking locally:-

1. If a decision table is read-only (old version, or locked by another User); I think you used to be able to look at the column definitions but not alter them. If the columns tab is now disabled we loose the ability to look at the columns' definitions when read-only.

2. If a decision table is pinned you should not be able to drag it.

**Comment by manstis** [comment]
Answering my own questions:-

1. Appears not!

2. This seems to be a bug and not a regression. ```GridWidgetDnDMouseDownHandler``` is putting DnD operations into "move table pending" state when the mouse pointer is over the "drag handle" (i.e. the bit with the table name it in) even if the Layer is in "pinned mode". This should not happen. Dragging by the "drag handle" re-positions the table on the canvas. This is different to "panning" that just moves the viewport. I'll open a JIRA.

**Comment by manstis** [comment]
For the record, the following JIRAs have been created:

- https://issues.jboss.org/browse/DROOLS-2140

- https://issues.jboss.org/browse/DROOLS-2141 (evident in this PR, but let's treat as a new JIRA!)

**Comment by jomarko** [comment]
@manstis thanks for the issues, I set myself as tester. Especially the second one proves your QE soul inside you.

**Comment by karreiro** [comment]
Jenkins please execute full downstream build.

**Comment by karreiro** [comment]
Jenkins please execute full downstream build.

**Comment by karreiro** [comment]
The broken tests are not related to the PR:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.462 s <<< FAILURE! - in org.jboss.errai.bus.server.websocket.test.jsr356.cdi.adapter.ContainerOnlyTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.209 s <<< FAILURE! - in org.jboss.errai.bus.server.websocket.test.jsr356.cdi.filter.FilterTest
```

**Pull request merged**